flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/taskmanager/TaskManagerInfo.java

```diff
@@ -113,7 +123,8 @@ public TaskManagerInfo(
             @JsonProperty(FIELD_NAME_TOTAL_RESOURCE) ResourceProfileInfo totalResource,
             @JsonProperty(FIELD_NAME_AVAILABLE_RESOURCE) ResourceProfileInfo freeResource,
             @JsonProperty(FIELD_NAME_HARDWARE) HardwareDescription hardwareDescription,
-            @JsonProperty(FIELD_NAME_MEMORY) TaskExecutorMemoryConfiguration memoryConfiguration) {
+            @JsonProperty(FIELD_NAME_MEMORY) TaskExecutorMemoryConfiguration memoryConfiguration,
+            @JsonProperty(FIELD_NAME_BLOCKED) @Nullable Boolean blocked) {
```

Comment:
With @JsonInclude(Include.NON_DEFAULT), the json may not include the blocked field when it's false. We add the annotation since we don't want to change the REST API reponse for non speculative execution users.
While when deserializing, jackson will try to apply a null to the field when the field is absent, and it will cause exception if the field is a boolean.